idx/memory/memory.go

```diff
@@ -242,17 +237,17 @@ func (m *MemoryIdx) add(def *schema.MetricDefinition) error {
 		log.Debug("memory-idx: creating branch %s with child %s", branch, nodes[i])
```

Comment:
That definitely wont work.

The purpose of this loop is to search backwards down the tree until we find a branch that exists.
then from https://github.com/raintank/metrictank/blob/9b2623b29398feb54795eb4bf0a9ecc42f544ba0/idx/memory/memory.go#L231-L243
we walk back up the tree creating all of the needed nodes.